### PR TITLE
statistics: avoid unnecessay table id loading the stats

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### What changed and how does it work?

1. avoid unnecessay table id loading the stats after the for-loop
2. smaller steps to accommodate a table scale of 1000+.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/statistics/handle/bootstrap.go [hunk 1]
```diff
@@ -40,6 +40,8 @@ import (
 	"go.uber.org/zap"
 )
 
+const initStatsStep = int64(500)
```
Comment (elsa0520) [inline pkg/statistics/handle/bootstrap.go:43]: Why was this value changed from the default 1000 to 500. What is the impact of changing the size?

Comment (hawkingrei) [reply pkg/statistics/handle/bootstrap.go:43]: I'm concerned that having a partitioned table within a single task might result in a very slow load time.

**Review by elsa0520 — APPROVED** [review]
LGTM

File: pkg/statistics/handle/bootstrap.go (hunk 1 above)
Comment (hi-rustin) [inline pkg/statistics/handle/bootstrap.go:43]: Please add some comments to this magic number. Thanks!

**Review by hi-rustin — APPROVED** [review]
/hold

nit: please add the comments

File: pkg/statistics/handle/bootstrap.go (hunk 1 above)
Comment (hawkingrei) [reply pkg/statistics/handle/bootstrap.go:44]: Thanks, I have updated it.

**Pull request merged**